pkg/apis/pipeline/v1beta1/param_types_test.go

```diff
@@ -168,23 +169,56 @@ type ArrayOrStringHolder struct {
 
 func TestArrayOrString_UnmarshalJSON(t *testing.T) {
 	cases := []struct {
-		input  string
+		input  map[string]interface{}
 		result v1beta1.ArrayOrString
 	}{
-		{"{\"val\": \"123\"}", *v1beta1.NewArrayOrString("123")},
-		{"{\"val\": \"\"}", *v1beta1.NewArrayOrString("")},
-		{"{\"val\":[]}", v1beta1.ArrayOrString{Type: v1beta1.ParamTypeArray, ArrayVal: []string{}}},
-		{"{\"val\":[\"oneelement\"]}", v1beta1.ArrayOrString{Type: v1beta1.ParamTypeArray, ArrayVal: []string{"oneelement"}}},
-		{"{\"val\":[\"multiple\", \"elements\"]}", *v1beta1.NewArrayOrString("multiple", "elements")},
+		{
+			input:  map[string]interface{}{"val": "123"},
+			result: *v1beta1.NewArrayOrString("123"),
+		},
+		{
+			input:  map[string]interface{}{"val": ""},
+			result: *v1beta1.NewArrayOrString(""),
+		},
+		{
+			input:  map[string]interface{}{"val": nil},
+			result: v1beta1.ArrayOrString{Type: v1beta1.ParamTypeArray, ArrayVal: nil},
```

Comment:
I know in current code we will unmarshal this to array, but I think this should be a string makes more sense?

Reply:
🤷 

That feels like a discussion to have in another PR - a null is ambiguous since it's neither a string or an array, but I can see why string might make more sense if we look at this as 1 value (string) vs many values (array).

The other option is we add enforcement to not allow null values, but all of this is outside the scope of what I wanted to do here 🙃

Reply:
yes, I added this change in my pr, just want to bring this up here for more thoughts! 😄